Scripts/application.py

```diff
@@ -14,99 +16,216 @@ def hello_world():
+                return(
+                'There was an error. Check that the requested ID is valid.')
+        return "user has been removed from database"
     
     if request.method == "GET":
         # returns a json of <person_id>
         with sqlite3.connect(path_to_db) as connection: 
             cursor = connection.cursor()
             try:
-                command = 'SELECT person_ID, email, givingTo_ID FROM people WHERE person_ID=?;'
+                command = 'SELECT person_ID, email, givingTo_ID FROM people ' \
+                          'WHERE person_ID=?;'
                 cursor.execute(command, (person_id,))
-                raw_data: List = cursor.fetchall()
-                data: Dict = {raw_data[0][0]:{"email": raw_data[0][1], "assignment ID": raw_data[0][2]}}
+                raw_data: Tuple = cursor.fetchone()
+                data: List = [
+                    {
+                        "person_ID": raw_data[0],
+                        "email": raw_data[1],
+                        "assignment ID": raw_data[2]}]
                 json_data = json.dumps(data)
             except:
-                return('There was an error. Check that the requested ID is valid.')
+                return(
+                    'There was an error. Check that the requested ID is valid.')
         return json_data
     
 @app.route('/api/v1/forbidden_pairing', methods = ["GET","POST"])
 def return_forbidden_pairings():
     
     if request.method == "POST":
-        new_giver_constraint = request.get_json()["santa"]
-        new_receiver_constraint = request.get_json()["assignment"]
+    
+        # verify that request contains a valid request  
+        try:
+            request_json = request.get_json()
+        except:
+            abort(
+                400,
+                description = "Please verify that request contains valid json.")
+        try:
+            new_giver_constraint = request_json['santa_ID']
+        except:
+            abort(
+                400,
+                description = "Please verify that request contains a "
+                              "'santa_ID' key.")
+        try:
+            new_receiver_constraint = request_json['assignment_ID']
+        except:
+            abort(
+                400,
+                description = "Please verify that request contains an "
+                              "'assignment_ID' key.")
+        
+        # add new constraint to database
         with sqlite3.connect(path_to_db) as connection:
             cursor = connection.cursor()
-            command = 'SELECT (person_ID) FROM people WHERE email = ?;'
+            command = 'SELECT (email) FROM people WHERE person_ID = ?;'
             cursor.execute(command, (new_giver_constraint,))
             giver_ID = cursor.fetchone()[0]
-            cursor.execute(command, (new_receiver_constraint,))
+            cursor.execute(command, (new_receiver_constraint,)) # finish swapping IDs for emails!!!

```

Comment:
A good convention for todo items is something like TODO JRodmanYoung: finish swapping IDs for emails. It's nice because then you can just search your code for TODO and find them.

Reply:
Makes sense.